service/cloudconfig/cloud_config.go

```diff
@@ -1,43 +1,267 @@
 func (we *WorkerExtension) Units() ([]k8scloudconfig.UnitAsset, error) {
-	return nil, nil
+	unitsMeta := []k8scloudconfig.UnitMetadata{
+		{
+			AssetContent: getKeyVaultSecretsUnit,
+			Name:         "get-keyvault-secrets.service",
+			Enable:       true,
+			Command:      "start",
+		},
+	}
+
+	units, err := we.renderUnits(unitsMeta)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return units, nil
 }
 
 // VerbatimSections allows sections to be embedded in the worker cloudconfig.
 func (we *WorkerExtension) VerbatimSections() []k8scloudconfig.VerbatimSection {
 	return nil
 }
+
+// getWorkerSecretsScript returns the script for downloading the worker TLS
+// certificates from Key Vault on startup.
+func (we *WorkerExtension) getWorkerSecretsScript() (k8scloudconfig.FileAsset, error) {
+	secrets := keyVaultSecrets{
+		VaultName: key.KeyVaultName(we.CustomObject),
+		Secrets: []keyVaultSecret{
+			// Calico client.
+			keyVaultSecret{
+				SecretName: key.SecretName(certificatetpr.CalicoComponent, certificatetpr.CA),
+				FileName:   "/etc/kubernetes/ssl/calico/client-ca.pem",
+			},
+			keyVaultSecret{
+				SecretName: key.SecretName(certificatetpr.CalicoComponent, certificatetpr.Crt),
+				FileName:   "/etc/kubernetes/ssl/calico/client-crt.pem",
+			},
+			keyVaultSecret{
+				SecretName: key.SecretName(certificatetpr.CalicoComponent, certificatetpr.Key),
+				FileName:   "/etc/kubernetes/ssl/calico/client-key.pem",
+			},
+			// Etcd client.
+			keyVaultSecret{
+				SecretName: key.SecretName(certificatetpr.EtcdComponent, certificatetpr.CA),
+				FileName:   "/etc/kubernetes/ssl/etcd/client-ca.pem",
+			},
+			keyVaultSecret{
+				SecretName: key.SecretName(certificatetpr.EtcdComponent, certificatetpr.Crt),
+				FileName:   "/etc/kubernetes/ssl/etcd/client-crt.pem",
+			},
+			keyVaultSecret{
+				SecretName: key.SecretName(certificatetpr.EtcdComponent, certificatetpr.Key),
+				FileName:   "/etc/kubernetes/ssl/etcd/client-key.pem",
+			},
+			// Kubernetes worker.
+			keyVaultSecret{
+				SecretName: key.SecretName(certificatetpr.WorkerComponent, certificatetpr.CA),
+				FileName:   "/etc/kubernetes/ssl/worker-ca.pem",
+			},
+			keyVaultSecret{
+				SecretName: key.SecretName(certificatetpr.WorkerComponent, certificatetpr.Crt),
+				FileName:   "/etc/kubernetes/ssl/worker-crt.pem",
+			},
+			keyVaultSecret{
+				SecretName: key.SecretName(certificatetpr.WorkerComponent, certificatetpr.Key),
+				FileName:   "/etc/kubernetes/ssl/worker-key.pem",
+			},
+		},
+	}
+	getSecretsScript, err := we.renderGetSecretsScript(secrets)
+	if err != nil {
+		return k8scloudconfig.FileAsset{}, microerror.Mask(err)
+	}
+
+	return getSecretsScript, nil
+}
+
+func (c *CloudConfigExtension) renderGetSecretsScript(secrets keyVaultSecrets) (k8scloudconfig.FileAsset, error) {
+	secretsMeta := k8scloudconfig.FileMetadata{
+		AssetContent: getKeyVaultSecretsTemplate,
+		Path:         "/opt/bin/get-key-vault-secrets",
```

Comment:
Also could be a constant. Uncertain.

Reply:
This can be a constant in `cloud_config.go` to get rid of the magic string. I'll change.

Reply:
Done.